Added !bal command

l-p-d opened the pull request.

l-p-d committed:
Added !bal command

IAmTeamML commented:
would recommend !nug balance as opposed to !bal in case if it conflicts with another bot

Commit:
Merge branch 'master' into lpd11-bal

xeoth commented on src/bot.py:
```diff
@@ -94,7 +94,14 @@ def int_conv(string: str) -> bool:
-
+            
+        # bal command
+        if re.match(r"!bal", comment.body): # as regex so we can add matches easily if needed
```
Won't this conflict with line 92? Won't the bot continue if it doesn't match `!(nug|nugget|gold)`?

l-p-d replied:
If people do two commands in one, i.e. `!bal !nug 4` it will. I think we should the regex to have to equal the message, rather than just be contained inside. Otherwise all other commands, i.e. the leaderboard one will have that issue, and people shouldn't be able to run multiple commands at once anyway.

max-landry replied:
This will conflict, the bot will reject the comment due to not being a nug comment before it gets processed. This should really be !nug bal as opposed to !bal for the reasons I said earlier

xeoth reviewed (changes requested):
We should either wrap the entire `!nug` code in a big `if` statement and make `!bal` into an `elif` or do as @IAmTeamML said

IAmTeamML commented:
Biased but I think !nug bal makes more sense conceptually and is less likely to summon a random bot, as well as avoids the earlier issue

Xeoth commented:
Lol we can just ban the other bot. I'd encourage wrapping it in `if`-`elif` to make making potential future commands simpler.

Xeoth commented:
And if we do that it could also resolve #24

Commit:
separated comments into if-elif

Commit:
altered comments and turned strings into fstrings

Commit:
changed every instance of 'minecraftmeme' into os.getenv

Commit:
removed pointless database

Xeoth commented:
With these changes you'll have to  add `SUBREDDIT=minecraftmeme` into `.env`.

Commit:
removed redundant check

Commit:
moved adding comment to DB higher up

l-p-d committed:
Minor additions

Changed starting nugs to 5, added `!bal` case for users not in database and slightly cleaned up code
